src/cpp/src/scheduler.hpp

```diff
@@ -167,6 +169,15 @@ class Scheduler {
         return std::numeric_limits<size_t>::max();
     }
 
+    void _restore_cached_blocks(const std::vector<SequenceGroup::Ptr>& sequence_groups) {
+        for (size_t sequence_group_id = 0; sequence_group_id < sequence_groups.size(); ++sequence_group_id) {
+            SequenceGroup::Ptr sequence_group = sequence_groups[sequence_group_id];
+            if (sequence_group->can_generate_tokens() || sequence_group->num_running_seqs() != 1)
```

Comment:
do we need a check for `sequence_group->num_running_seqs() != 1` ? 
Looks like we cannot generate new tokens (we are in prompt phase), then we have the only sequence within a group